### PR TITLE
GTT: 1071 - Added static datasets selection page

### PR DESCRIPTION
**Pull request by ajmokotoff**
## Description

Added static datasets selection page to add charts

## Testing

Wrote unit tests and verified it. See it live here: https://drjun54o2bnsn.cloudfront.net/

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/components/Table.tsx [hunk 1]
```diff
@@ -282,4 +302,27 @@ const IndeterminateCheckbox = React.forwardRef<
   return <input type="checkbox" title={title} ref={resolvedRef} {...rest} />;
 });
 
+const IndeterminateRadio = React.forwardRef<
+  HTMLInputElement,
+  {
+    indeterminate?: boolean;
+    title?: string;
+    checked?: boolean;
+    onClick?: any;
```
Comment (ferdingler) [inline frontend/src/components/Table.tsx]: make the `onClick` type `Function`?

File: frontend/src/containers/ChooseStaticDataset.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    history.goBack();
+  };
+
+  const onSelect = useCallback((selectedDataset: Array<Dataset>) => {
+    setSelected(selectedDataset);
+  }, []);
+
+  const onSubmit = () => {
+    history.replace({
+      pathname: `/admin/dashboard/${dashboardId}/add-chart`,
```
Comment (ferdingler) [inline frontend/src/containers/ChooseStaticDataset.tsx]: How are we going to reuse this screen for other types of content items? I think this screen needs to be agnostic of what type of content item the user is adding. Should be easy to do though, just have this screen receive the `redirectUrl` as part of the state, and then when the user selects a dataset, you redirect them back to where they came from.

File: frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  });
+  expect(
+    screen.getByRole("button", { name: "Select and continue" })
+  ).toBeInTheDocument();
+});
+
+test("renders a cancel button", () => {
+  render(<ChooseStaticDataset />, {
+    wrapper: MemoryRouter,
+  });
```
Comment (ferdingler) [inline frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx]: Nit pick: This render instruction is repeated on every test. You can extract it into a beforeEach: 

```js
beforeEach(() => {
  render(<ChooseStaticDataset />, {
    wrapper: MemoryRouter,
  });
});
```

File: frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+test("renders search box", () => {
+  render(<ChooseStaticDataset />, {
+    wrapper: MemoryRouter,
+  });
+  expect(screen.getByRole("searchbox", { name: "Search" })).toBeInTheDocument();
+});
+
+test("renders a selet and continue button", () => {
```
Comment (ferdingler) [inline frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx]: typo in `select`

File: frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx (hunk 3 above)
Comment (ferdingler) [reply frontend/src/containers/__tests__/ChooseStaticDataset.test.tsx]: Adding another test to verify that a Table is rendered with the dummy datasets from your mocked hook would be good too.

**Review by ferdingler — CHANGES_REQUESTED** [review]

File: frontend/src/containers/ChooseStaticDataset.tsx (hunk 2 above)
Comment (ajmokotoff) [reply frontend/src/containers/ChooseStaticDataset.tsx]: That is a very good point. Let me fix that.

**Review by ferdingler — APPROVED** [review]

**Pull request merged**